### PR TITLE
Adding azure fileshare backup template with hourly backup policy

### PR DESCRIPTION
**Pull request by kumarisarita5**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Adding azure fileshare backup bicep template with hourly backup policy

### CODE REVIEW

File: quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+param isNewVault bool = true
+
+@description('Set to true if a new Backup Policy is to be created for the Recovery Services Vault; set to false otherwise.')
+param isNewPolicy bool = true
+
+@description('Set to true if the existing Storage Account has to be registered to the Recovery Services Vault; set to false otherwise.')
+param registerStorageAccount bool = true
+
+@description('Name of the Recovery Services Vault. (Should have the same location as the Storage Account containing the File Share to be protected in case of an existing Recovery Services Vault).')
+param vaultName string = 'RSVault'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep:23]: us a globally unique defaultValue with uniqueString()

File: quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+resource protectionContainer 'Microsoft.RecoveryServices/vaults/backupFabrics/protectionContainers@2021-12-01' = if (registerStorageAccount) {
+  name: '${vaultName}/${backupFabric}/storagecontainer;Storage;${existingResourceGroupName};${existingStorageAccountName}'
+  properties: {
+    backupManagementType: backupManagementType
+    containerType: 'StorageContainer'
+    sourceResourceId: resourceId(existingResourceGroupName, 'Microsoft.Storage/storageAccounts', existingStorageAccountName)
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep:159]: move dependsOn above properties

File: quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+resource protectedItem 'Microsoft.RecoveryServices/vaults/backupFabrics/protectionContainers/protectedItems@2021-12-01'={
+  name:'${split('${vaultName}/${backupFabric}/storagecontainer;Storage;${existingResourceGroupName};${existingStorageAccountName}', '/')[0]}/${split('${vaultName}/${backupFabric}/storagecontainer;Storage;${existingResourceGroupName};${existingStorageAccountName}', '/')[1]}/${split('${vaultName}/${backupFabric}/storagecontainer;Storage;${existingResourceGroupName};${existingStorageAccountName}', '/')[2]}/AzureFileShare;${existingFileShareName}'
+  properties:{
+    protectedItemType:'AzureFileShareProtectedItem'
+    sourceResourceId:resourceId(existingResourceGroupName, 'Microsoft.Storage/storageAccounts', existingStorageAccountName)
+    policyId:backupPolicy.id
+  }
+  dependsOn:[
```
Comment (bmoore-msft) [inline quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep:172]: move dependsOn above properties

File: quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+@description('Number of years for which the yearly backup is to be retained. (Ignore if using existing Backup Policy).')
+param yearlyRetentionDurationCount int = 10
+
+@description('Hourly Schedule window start time')
+param scheduleWindowStartTime string = '${substring(utcNow('2020-01-01T{0}:00Z'), 0, 11)}08:00:00.000Z'
+
+var backupFabric = 'Azure'
+var backupManagementType = 'AzureStorage'
+var scheduleRunTimes = [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep:65]: this doesn't seem to be used to schedule the backup time - only the retention time... is that expected?

File: quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+resource backupPolicy 'Microsoft.RecoveryServices/vaults/backupPolicies@2022-02-01' = if (isNewPolicy) {
+  parent: vault
+  name: policyName
+  properties: {
+    backupManagementType: backupManagementType
+    schedulePolicy: {
+      schedulePolicyType: 'SimpleSchedulePolicy'
+      scheduleRunFrequency: 'Hourly'
+      scheduleRunDays: null
```
Comment (bmoore-msft) [inline quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/main.bicep:88]: remove null properties

File: quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/metadata.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "itemDisplayName": "Backup existing File Share using Recovery Services",
```
Comment (bmoore-msft) [inline quickstarts/microsoft.recoveryservices/recovery-services-backup-file-share-hourly-backup/metadata.json:3]: there is already a sample in the repo with this metadata

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
also check the pipeline results for changes that need to be made

**Comment by kumarisarita5** [comment]
Hi @bmoore-msft, without the prereq.azuredeploy.json file the deployment checks are failing, So I've kept this file in the PR.
I've removed the azuredeploy.json and will add it later in new PR.

**Comment by kumarisarita5** [comment]
Hi @bmoore-msft, could you please review this PR

**Pull request merged**